iothub/device/src/Transport/Mqtt/MqttIotHubAdapter.cs

```diff
@@ -386,66 +342,51 @@ private async Task ConnectAsync(IChannelHandlerContext context)
                 }
                 _stateFlags = StateFlags.Connecting;
 
-                ScheduleCheckConnectTimeout(context);
+                await ScheduleCheckConnectTimeoutAsync(context).ConfigureAwait(true);
```

Comment:
**nit:** All of these ConfigureAwait(true) calls can be removed to make the code less verbose. Unless of course FxCop yells about it the I guess it's okay.

see: https://devblogs.microsoft.com/dotnet/configureawait-faq/#why-would-i-want-to-use-configureawaittrue

Reply:
Not a nit at all, actually. Typically this would be very bad.

However, and it is my fault for not pointing this out, at the top of this file there is this comment:

```csharp
     // Note on ConfigureAwait: dotNetty is using a custom TaskScheduler that binds Tasks to the corresponding
    // EventLoop. To limit I/O to the EventLoopGroup and keep Netty semantics, we are going to ensure that the	    // EventLoop. To limit I/O to the EventLoopGroup and keep Netty semantics, we are going to ensure that the
    // task continuations are executed by this scheduler using ConfigureAwait(true).	    // task continuations are executed by this scheduler using ConfigureAwait(true).
```

Reply:
As a rule, we use `ConfigureAwait(false)` on all library async calls. This appears to be an exception.